docs/core/diagnostics/diagnostics-in-containers.md

```diff
@@ -0,0 +1,69 @@
+
+```dockerfile
+# In build stage
+# Install desired .NET CLI diagnostics tools
+RUN dotnet tool install --tool-path /tools dotnet-trace
+RUN dotnet tool install --tool-path /tools dotnet-counters
+RUN dotnet tool install --tool-path /tools dotnet-dump
+
+...
+
+# In final stage
+# Copy diagnostics tools
+WORKDIR /tools
+COPY --from=build /tools .
+```
+
+Alternatively, the .NET Core SDK can be installed in a container when needed in order to install the CLI tools. Be aware that installing the .NET Core SDK will have the side-effect of reinstalling the .NET Core runtime. So be sure to install the version of the SDK that matches the runtime present in the container.
+
+### Using .NET Core global CLI tools in a sidecar container
+
+If you would like to use .NET Core global CLI diagnostic tools to diagnose processes in a different container, bear the following additional requirements in mind:
+
+1. The containers must, of course, [share a process namespace](https://docs.docker.com/engine/reference/run/#pid-settings---pid) (so that tools in the sidecar container can access processes in the target container).
+2. The .NET Core global CLR diagnostic tools need access to files the .NET Core runtime writes to the /tmp directory, so the /tmp directory must be shared between the target and sidecar container via a volume mount. If you attempt to use the diagnostic tools from a sidecar container without sharing the /tmp directory, you will get an error about the process "not running compatible .NET runtime."
+
+## Using `PerfCollect` in a container
+
+The [`PerfCollect`](https://github.com/dotnet/coreclr/blob/master/Documentation/project-docs/linux-performance-tracing.md) script is useful for collecting performance traces and is the recommended tool for collecting traces prior to .NET Core 3.0. If using `PerfCollect` in a container, keep the following requirements in mind:
+
+1. `PerfCollect` requires the [`SYS_ADMIN` capability](https://man7.org/linux/man-pages/man7/capabilities.7.html) (in order to run the `perf` tool), so be sure the container is [started with that capability](https://docs.docker.com/engine/reference/run/#runtime-privilege-and-linux-capabilities).
+2. `PerfCollect` requires that the following environment variables be set prior to the app it is profiling starting. Because Docker entrypoints start almost immediately, it's necessary to define these environment variables when starting the container.
```

Comment:
I found this unnecessarily confusing. I'd rather see a sample Dockerfile (like in the case of installing the CLI tools) that sets these variables

Reply:
These may not be in a Dockerfile, though, because customers may set the env vars in a K8s manifest or in some other way so that they're defined when starting the container. I'll try to rephrase this section to make it clearer.